Dockerfile

```diff
@@ -19,10 +21,17 @@ RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s
     chmod +x ./kubectl && \
     mv ./kubectl /usr/local/bin
 
-#cloning the gluster-mixins project and making extras as working dir
-RUN git clone https://github.com/gluster/gluster-mixins /gluster/gluster-mixins;
-WORKDIR /gluster/gluster-mixins/extras
+#cloning the gluster-mixins project and making gluster-mixins as working dir
+COPY . /gluster/gluster-mixins/
+WORKDIR /gluster/gluster-mixins/
+
+#run test and generate the intermidiate files
+RUN make test \ 
```

Comment:
`RUN make` should be sufficient as it will generate intermediate files, run `promtool` check on it and then run `tests`.

Reply:
If we do `RUN make `it run the tests then why to run promtool again? 

My Idea behind this was to make sure the make test is passing before anything gets generated. what you thing?

I can change to RUN make only.